scipy/linalg/_decomp_polar.py

```diff
@@ -99,6 +99,17 @@ def polar(a, side="right"):
     a = np.asarray(a)
     if a.ndim != 2:
         raise ValueError("`a` must be a 2-D array.")
+    # accommodate square empty matrices
+    if a.size == 0:
+        dt = polar(np.eye(2, dtype=a.dtype))[0].dtype
```

Comment:
Same with the other PR. This is way too heavy to compute to handle empty arrays. Use numpy.result_type and upcast integer results to `np.float64`